.github/workflows/build-and-publish-search-index.yml

```diff
@@ -0,0 +1,27 @@
+name: Build And Publish Search Index
+
+on:
+  push:
+    branches:
+      - main
```

Comment:
If this is published on only push to main, our PR previews's search won't be based on that content but the main content.

Perhaps this is acceptable.. is there a way to have multiple indexes?

Reply:
We can have multiple indicies, but you can't just create them on the fly. So if we had 2 indicies, then if person A creates PR 1 and person B creates PR 2, then they both upsert into the same dev index, then PR 1's preview will be querying an index that's not representative of their current preview anyway

Reply:
In that case, it seems reasonable for now to just only use and update the index for the prod version of our site.

Reply:
The other potential problem I see with this is that since the publishing of index is separate from our site cloud build, they can fail independently causing the site and index to be out of sync. The danger is quite minimal since I wouldn't expect the site content to have such a significant change that the off sync would be much noticeable..

We could potentially move the index upload into the `cloudbuild-main.yaml` as an extra step so it happens together with the site build instead of within github actions. It's possible to add a step like this https://github.com/lit/lit.dev/blob/de6e81d70a123a6ceb76471ed67b43d8b714c4c2/cloudbuild-cleanup.yaml#L6-L12
Maybe to run after the site build is successful? Then if the site build/deploy fails, we don't upload the new index. The algolia write key would need to be added to GCP then.

Reply:
I believe in you

Reply:
I will not fail you!

Reply:
fwiw, I think it's fine to have the index only be for the live site.

@augustjk being part of the build sounds probably a bit better, but given that Cloud Run lets us deploy separately from builds, there's a chance that's not quite right either. I think it's an edge case of an edge case, but it'd be cool if the index was updated on _deployment_ of an image.

Reply:
The cloud build is responsible for both building and deploying so adding a step after that for the index upload would work. If we manually change which revision gets the traffic, this potentially won't work but that would be a very rare occurrence.

Now that I think about it more though, I wonder if it's better to keep this in github actions and configure the upload to be a wireit script so we can take advantage of wireit cache and skip unless the index actually changed?

We don't have a solution for utilizing wireit cache for cloud build yet. It's possible to manually configure persistence of artifacts with google storage. Would it be worth it to pursue that?

Reply:
created wireit command